common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go

```diff
@@ -0,0 +1,1086 @@
+				query.EXPECT().Consistency(cassandraAllConslevel).Return(query).Times(1)
+				query.EXPECT().MapScanCAS(gomock.Any()).DoAndReturn(func(prev map[string]interface{}) (bool, error) {
+					return false, errors.New("failed to delete")
+				}).Times(1)
+			},
+			clientMockFn: func(client *gocql.MockClient) {
+				client.EXPECT().IsCassandraConsistencyError(gomock.Any()).Return(false).Times(1)
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			query := gocql.NewMockQuery(ctrl)
+			tc.queryMockFn(query)
+			session := &fakeSession{
+				query: query,
+			}
+			client := gocql.NewMockClient(ctrl)
+			if tc.clientMockFn != nil {
+				tc.clientMockFn(client)
+			}
+			cfg := &config.NoSQL{}
+			logger := testlogger.New(t)
+			dc := &persistence.DynamicConfiguration{}
+
+			db := newCassandraDBFromSession(cfg, session, logger, dc, dbWithClient(client))
+
+			err := db.DeleteTaskList(context.Background(), tc.filter, tc.prevRangeID)
+
+			if (err != nil) != tc.wantErr {
+				t.Errorf("DeleteTaskList() error = %v, wantErr %v", err, tc.wantErr)
+			}
+
+			if err != nil {
+				return
+			}
+
+			if diff := cmp.Diff(tc.wantQueries, session.queries); diff != "" {
+				t.Fatalf("Query mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestInsertTasks(t *testing.T) {
+	ts, err := time.Parse(time.RFC3339, "2024-04-01T22:08:41Z")
+	if err != nil {
+		t.Fatalf("Failed to parse time: %v", err)
+	}
+
+	tests := []struct {
+		name                      string
+		tasksToInsert             []*nosqlplugin.TaskRowForInsert
+		tasklistCond              *nosqlplugin.TaskListRow
+		mapExecuteBatchCASApplied bool
+		mapExecuteBatchCASErr     error
+		mapExecuteBatchCASPrev    map[string]any
+		wantQueries               []string
+		wantErr                   bool
+	}{
+		{
+			name: "successfully applied",
+			tasksToInsert: []*nosqlplugin.TaskRowForInsert{
+				{
+					TTLSeconds: 0, // default create task query will be used for this ttl
+					TaskRow: nosqlplugin.TaskRow{
+						TaskID:      3,
+						WorkflowID:  "wid1",
+						RunID:       "rid1",
+						ScheduledID: 42,
+					},
+				},
+				{
+					TTLSeconds: int(maxCassandraTTL) + 1, // ttl query will be used for this ttl
+					TaskRow: nosqlplugin.TaskRow{
+						TaskID:      4,
+						WorkflowID:  "wid1",
+						RunID:       "rid1",
+						ScheduledID: 43,
+					},
+				},
+			},
+			tasklistCond: &nosqlplugin.TaskListRow{
+				DomainID:     "domain1",
+				TaskListName: "tasklist1",
+				TaskListType: 1,
+				RangeID:      25,
+			},
+			mapExecuteBatchCASApplied: true,
+			wantQueries: []string{
+				`INSERT INTO tasks (domain_id, task_list_name, task_list_type, type, task_id, task) VALUES(domain1, tasklist1, 1, 0, 3, {domain_id: domain1, workflow_id: wid1, run_id: rid1, schedule_id: 42,created_time: 0001-01-01T00:00:00Z, partition_config: map[] })`,
```

Comment:
the `created_time`, is that expected to be zero?

Reply:
I missed to pass `TaskRow.CreatedTime`. Good catch let me add that